### ISSUE TITLE
Let each entry choose which sites it should be saved to - Issue with Revisions

### ISSUE THREAD

**Issue by bkrsmanovic** [craftcms/cms#9844]
### Description
When you have structure / single with "**Let each entry choose which sites it should be saved to**" settings, site admins that are in charge of specific sites where entry is enabled to, can see revisions from same entry on other sites. It causes numerous of issues with activating specific revision specifically from other users on other sites.

### Steps to reproduce

1. Create page on Site 1:
![Screenshot from 2021-09-16 11-40-53](https://user-images.githubusercontent.com/15169280/133589584-1f1e0478-b70f-4600-973a-3b7e823188e5.png)

2. Enable that entry on Site 2:
![Screenshot from 2021-09-16 11-12-33](https://user-images.githubusercontent.com/15169280/133589620-fbbbaba9-4c0c-4bf4-8814-d9d40b919f45.png)
Name new enabled page on Site 2: Testing Page Site 2 (for distinguish them).

3. Do couple of edits (for example Title of page) with **global_admin** (or **site1_admin**) on Testing Page Site 1, and in meanwhile after every edit, do edit with **site2_admin** on Testing Page Site 2, change title to something, just so you can have mixed list of revisions:
![Screenshot from 2021-09-16 11-47-42](https://user-images.githubusercontent.com/15169280/133590629-87f25486-b711-4975-b65b-d96b627dd8bd.png)
Now for example, **site2_admin** on **Testing Page Site 2**, can activate any Revision from **global_admin** or **site1_admin**, which are done on **Testing Page Site 1**, and when doing that, it changes content on Testing Page Site 1 to that revision + sometimes it doesnt apply proper changes to Testing Page Site 2, you cant even see what's change gonna be, example:
![Screenshot from 2021-09-16 11-49-49](https://user-images.githubusercontent.com/15169280/133590963-10f793a3-ba2d-4b3a-978d-97b28625082b.png)
(Testing Page Site 2 Name 2 is not revision from **global_admin**).

This is just issues with Title, there are tons of other issues with other content, matrix fields, assets etc.

TLDR:
(a) I think admins that are in charge of specific sites should not see revisions from other admins on other entries on other sites.
(b) Maybe even restrict revisions on per-entry base, so even if you enable it for other site, with all-access admin, revisions should kept on entry level, not cross-site.
If they have to be cross-site, then (a) should be implemented.

### Additional info

- Craft version: Craft CMS 3.7.12
- PHP version: 8
- Database driver & version: MariaDB
- Plugins & versions:

**Comment by brandonkelly** [comment]
> When you have structure / single with "**Let each entry choose which sites it should be saved to**" settings, site admins that are in charge of specific sites where entry is enabled to, can see revisions from same entry on other sites.

There is no such thing as site-specific revisions. Whenever you change content in one site, it may have effects on the other sites’ content, per the fields’ **Translation Method** settings. (If a field is not translatable, its new value will be copied to all other sites, for example). So each time you save an entry, it will get a new revision that stores the entry’s state across each of its sites.

Likewise, when you revert an entry to a prior revision, all of its content from that prior revision will be reverted; not just the site you are looking at.
